controllers/admin/projects/group_allocations/group_allocations.controller.js

```diff
@@ -0,0 +1,73 @@
+const model = require('../../../../models/index')
+
+// Return all GroupAllocations of this project
+exports.findAll = (req, res) => {
+  model.Group.findAll({ where: { project_id: req.params.project_id }})
+  .then(groups => {
+    const group_ids = groups.map(group => group.id);
+
+    model.GroupAllocation.findAll({ where: { group_id: group_ids }})
+      .then(group_allocations => {
+        res.send(group_allocations);
+      })
+      .catch(err => {
+        res.status(500).send({
+          message:
+            err.message || `Some error occurred while retrieving Group Allocations for Project ${req.params.project_id}`
+        });
+      });
+  })
+  .catch(err => {
+    res.status(500).send({message: err.message || `Error occurred retrieving Groups for Project ${req.params.project_id}`});
+  })
+}
+
+// Create a Group Allocation associated to this Project
+exports.createOne = (req, res) => {
+  if (req.body.group_id && req.body.external_id) {
+    model.User.findOne({ where: { external_id: req.body.external_id }}).then(user => {
```

Comment:
There is no constraint that there can't be 2 users with the same `external_id` so I think this should be changed to just `id` at all the places unless we want to add that as a constraint on the model (which I think would be a better idea given how much `external_id` has been used.

Reply:
Also, right now, a user can be added to a group multiple times. You can have a group of 5 with all 5 being the same guy. We can either fix this or frankly just ignore it and never demonstrate that in the final presentation, they will have no way of knowing, or we can just leave to resolve this at the end and write it down in the ISSUES section of the repo.

Reply:
>  write it down in the ISSUES section of the repo

Yep please do that, just copy your comments across